americangut/results_utils.py

```diff
@@ -595,3 +597,144 @@ def write_bloom_fasta(unique_counts, output_file, abundance_threshold):
+        group : str
+            The value of the `group_field` for the sample
+        group_alpha : ndarray
+            The alpha diversity values associated with the group
+        sample_alpha : float
+            The alpha diversity for the sample
+        xlabel : str
+            The label used for the x-axis of the plot.
+
+    """
+
+    # Explicitly casts the alpha diversity to a float
+    alpha_map[alpha_field] = alpha_map[alpha_field].astype(float)
+
+    # Draws the observations and group
+    group = alpha_map.loc[sample, group_field]
+    sample_alpha = alpha_map.loc[sample, alpha_field]
+    group_alpha = alpha_map.loc[alpha_map[group_field] == group, alpha_field]
+    if categorical:
+        # Keep only barcoded items, which can be converted to float.
+        # Categories will be strings in the index
+        def test(val):
+            try:
+                float(val)
+                return True
+            except ValueError:
+                return False
+        group_alpha = group_alpha.loc[[test(v) for v in group_alpha.index]]
+
+    if xlabel is None:
+        xlabel = '%sdiversity' % alpha_field.split('1')[0].replace('_', ' ')
+
+    if debug:
+        return group, group_alpha, sample_alpha, xlabel
+
+    # Defines the group color. This is currently hardcoded, although the
+    # longer term plan is to substitute in function which will define the color
+    # based on the relationship between the sample and a yet to be written
+    # predicted value.
+    group_color = '#1f78b4'
+
+    with sn.axes_style('ticks', {'axes.facecolor': 'none'}):
+        # Sets up the axis for plotting
+        ax = plt.axes()
+
+        # Plots the distribution
+        sn.kdeplot(group_alpha,
+                   ax=ax,
+                   legend=False,
+                   color=group_color)
+        ylim = ax.get_ylim()
+
+        # Plots the individual line
+        ax.plot([sample_alpha, sample_alpha], [-1, 1], color=sample_color)
+
+        # Returns the y-limits to the original value and removes the ticks
+        ax.set_ylim(ylim)
+        ax.set_yticks([])
+        # Removes the spine
+        sn.despine(offset=5, trim=True, top=True, left=True, right=True)
+        # Updates the xticks to match the correct font
+        ax.set_xticklabels(map(int, ax.get_xticks()), size=11)
+        ax.set_xlabel(xlabel, size=13)
+
+        # Highlight range, if given
+        if highlight_range is not None:
+            xticks = ax.get_xticks()
+            x, y = ax.lines[0].get_xydata().T
+            m = (highlight_range[0] < x) & (x < highlight_range[1])
+            ax.fill_between(x[m], 0, y[m], alpha=.15, color=group_color)
+            # Fill between changes axis ever so slightly, so need to reset
+            # back to what we want.
+            ax.set_xticks(xticks)
+
+        # Adds text describing the sample or category
+        if categorical:
+            text = '\n\n%s:\t%1.1f' % (sample, sample_alpha)
```

Comment:
The .png graphs that I generated contain the ```[]``` symbol after the colon. Perhaps the '\t' isn't being read properly?